client/components/QuestionsAndAnswers/QuestionItem.jsx

```diff
@@ -1,19 +1,28 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import {
   QuestionDiv, QuestionBody, QuestionMisc, UnderLine, LoadMore,
 } from './styles/QuestionItem.style';
 import AnswerList from './AnswerList';
 import AnswerModal from './modal/AnswerModal';
+import { postAnswer, updQuestionHelpfulness, getQuestions } from './helpers/helpers';
+import { useProduct } from '../../ProductContext';
 
-const QuestionItem = ({ question, index, updateHelpfulness }) => {
+const QuestionItem = ({
+  question, index, updateHelpfulness,
+}) => {
   const sortedAnswers = Object.values(question.answers).sort(
     (a, b) => b.helpfulness - a.helpfulness,
   );
   const [isAdd, setIsAdd] = useState(false);
   const [isHelpful, setIsHelpful] = useState(false);
   const [length, setLength] = useState(2);
   const [displayedAnswers, setDisplayedAnswers] = useState(sortedAnswers.slice(0, length));
+  const currProduct = useProduct();
+
+  // useEffect(() => {
+  //   setDisplayedAnswers(sortedAnswers.slice(0, setLength(sortedAnswers.length)));
+  // }, [question]);
 
   const loadMoreAnswers = (e) => {
```

Comment:
I think what you're doing in the else statement is a bit too hard to follow since it is 3 nested calls.